cli/cli.py

```diff
@@ -0,0 +1,306 @@
+}
+
+ALGOS = {
+    Algorithm.pyro: desbordante.Pyro,
+    Algorithm.tane: desbordante.Tane,
+    Algorithm.hyfd: desbordante.HyFD,
+    Algorithm.fd_mine: desbordante.FdMine,
+    Algorithm.dfd: desbordante.DFD,
+    Algorithm.dep_miner: desbordante.Depminer,
+    Algorithm.fdep: desbordante.FDep,
+    Algorithm.fun: desbordante.FUN,
+    Algorithm.fastfds: desbordante.FastFDs,
+    Algorithm.aid: desbordante.Aid,
+    Algorithm.naive_fd_verifier: desbordante.FDVerifier,
+    Algorithm.naive_afd_verifier: desbordante.FDVerifier,
+    Algorithm.naive_mfd_verifier: desbordante.MetricVerifier
+}
+
+
+def get_algorithm(ctx, param, value):
+    if value is not None:
+        return value
+    task = ctx.params.get('task')
+    if task is None:
+        raise click.MissingParameter(ctx=ctx, param=param)
+    return TASK_INFO[task].default
+
+
+def help_callback_func(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    check_mismatch(ctx.params.get('task'), ctx.params.get('algo'))
+    print_help_page(ctx.params.get('algo'), ctx.params.get('task'))
+    ctx.exit()
+
+
+def check_mismatch(task, algo):
+    if (task is not None and
+            (algo is not None and algo not in TASK_INFO[task].algos)):
+        click.echo(f"ERROR: Wrong value for 'algo' when task={task}.")
+        sys.exit(1)
+
+
+def check_error_option_presence(task, error):
+    if task in (Task.afd, Task.afd_verification) and error is None:
+        click.echo("ERROR: Missing option 'error'.")
+        sys.exit(1)
+    if task in (Task.fd, Task.fd_verification) and error is not None:
+        click.echo("ERROR: Invalid option: 'error'.")
+        sys.exit(1)
+
+
+def is_omitted(value):
+    return value is None or (type(value) is tuple and len(value) == 0)
+
+
+def set_option(algo, opt_name, opt_value):
+    try:
+        algo.set_option(opt_name, opt_value)
+    except Exception as exc:
+        click.echo(exc)
+        sys.exit(1)
+
+
+def set_algo_options(algo, args):
+    used_options = set()
+    while opts := algo.get_needed_options():
+        for option_name in opts:
+            if is_omitted(args[option_name]):
+                set_option(algo, option_name, None)
+            else:
+                set_option(algo, option_name, args[option_name])
+                used_options.add(option_name)
+    return used_options
+
+
+def get_algo_result(algo, algo_name):
+    try:
+        algo.execute()
+        match algo_name:
+            case Algorithm.naive_fd_verifier:
+                result = algo.fd_holds()
+            case Algorithm.naive_afd_verifier:
+                result = algo.get_error()
+            case Algorithm.naive_mfd_verifier:
+                result = algo.mfd_holds()
+            case algo_name if algo_name in TASK_INFO[Task.fd].algos:
+                result = algo.get_fds()
+            case _:
+                assert False, 'No matching get_result function.'
```

Comment:
```suggestion
                raise Exception('No matching get_result function.')
```
Тут есть такой момент
https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement
> These equivalences assume that [__debug__](https://docs.python.org/3/library/constants.html#debug__) and [AssertionError](https://docs.python.org/3/library/exceptions.html#AssertionError) refer to the built-in variables with those names. In the current implementation, the built-in variable [__debug__](https://docs.python.org/3/library/constants.html#debug__) is True under normal circumstances, False when optimization is requested (command line option [-O](https://docs.python.org/3/using/cmdline.html#cmdoption-O)). The current code generator emits no code for an assert statement when optimization is requested at compile time. Note that it is unnecessary to include the source code for the expression that failed in the error message; it will be displayed as part of the stack trace.

то есть это потенциально может быть багом, ну и плюс слегка излишне

Reply:
Всё правильно. Такой ситуации никогда не должно происходить, тут это явно обозначено

Reply:
Вообще говоря, по смыслу здесь должен быть как раз `assert`, т.к. такой ситуации не предполагается вовсе.

Reply:
```python
#The simple form, assert expression, is equivalent to

if __debug__:
    if not expression: raise AssertionError
```
Получается, что сейчас в коде что-то типа:
```python
if __debug__:
  if not False: raise AssertionError('No matching get_result function.')
```
Что ну так себе из-за ```if not False```. Ну и в любом случае мы получим raise исключения, но, если кто-то добавит -O, то мы получим ещё и баг, поскольку не будет выброшено исключение совсем

Reply:
Не понял. Почему имеет значение, что `assert` эквивалентен чему-то? В плюсовой части же мы пишем `assert` в некоторых местах --- этот код тоже в релизной версии не выполняется.

Если написано `assert False`, то это означает, что эта строка не должна выполняться никогда. Тут так и есть.